implementations/rust/ockam/ockam_command/src/manpages.rs

```diff
@@ -0,0 +1,140 @@
+use crate::help;
+use crate::OckamCommand;
+use clap::builder::NonEmptyStringValueParser;
+use clap::{ArgAction, Args, Command, CommandFactory};
+use clap_mangen::Man;
+use flate2::{Compression, GzBuilder};
+use std::fs::{create_dir, File};
+use std::io::{Error, Write};
+use std::path::{Path, PathBuf};
+use std::{env, io};
+use tracing::error;
+
+const LONG_HELP: &str = "\
+man pages output directory. Absolute path required. Will be created in case not existing. \
+Default: users home directory \"~/local/.share/man/man1/\". \
+Fallback: \"ockam_man_pages/\" in the current working directory.";
+
+/// Generate Ockam man pages
+#[derive(Clone, Debug, Args)]
+#[command(hide = help::hide())]
+pub struct ManpagesCommand {
+    #[arg(
+        short,
+        long,
+        help = "man pages output directory path",
+        long_help = LONG_HELP,
+        value_parser(NonEmptyStringValueParser::new())
+    )]
+    dir: Option<String>,
+
+    #[arg(
+        short,
+        long,
+        default_value = "false",
+        action = ArgAction::SetTrue,
+        help = "disable gzip compression for man page output",
+    )]
+    no_compression: bool,
+}
+
+impl ManpagesCommand {
+    pub fn run(self) {
+        write_man_pages(&self.dir, &self.no_compression);
+    }
+}
+
+fn write_man_pages(dir: &Option<String>, no_compression: &bool) {
+    let man_dir = match get_man_page_directory(dir) {
+        Ok(path) => path,
+        Err(error) => panic!("Error getting man page directory: {:?}", error),
+    };
+    let clap_command = <OckamCommand as CommandFactory>::command();
+    generate_man_pages(man_dir.as_path(), &clap_command, None, no_compression);
+}
+
+fn get_man_page_directory(cmd_man_dir: &Option<String>) -> io::Result<PathBuf> {
+    let man_dir = match cmd_man_dir {
+        Some(dir) => {
+            let mut user_specified_dir = PathBuf::new();
+            user_specified_dir.push(dir);
+            user_specified_dir
+        }
+        None => match dirs::home_dir() {
+            Some(mut home_dir) => {
+                home_dir.push(".local/share/man/man1/");
+                home_dir
+            }
+            None => {
+                let mut man_dir = env::current_dir()?;
+                man_dir.push("ockam_man_pages/");
+                man_dir
+            }
+        },
+    };
+
+    println!("man pages directory: {}", man_dir.display());
+    if !man_dir.exists() {
+        create_dir(man_dir.clone())?;
+    }
```

Comment:
Nit: you can replace this with `create_dir_all`, as you don't have to check if it already exists.

Reply:
Switched.